pkg/framework/webhooks.go

```diff
@@ -0,0 +1,149 @@
+	return wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Delete(context.TODO(), webhookConfiguraiton); err != nil {
+			klog.Errorf("error querying api for MutatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// UpdateMutatingWebhookConfiguration updates the specified mutating webhook configuration
+func UpdateMutatingWebhookConfiguration(c client.Client, updated *admissionregistrationv1.MutatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitMedium, func() (bool, error) {
+		existing, err := GetMutatingWebhookConfiguration(c, updated.Name)
+		if err != nil {
+			klog.Errorf("Error getting MutatingWebhookConfiguration: %v", err)
+			return false, nil
+		}
+		if err := c.Patch(context.TODO(), existing, client.MergeFrom(updated)); err != nil {
+			klog.Errorf("error patching MutatingWebhookConfiguration object %q: %v, retrying...", updated.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// UpdateValidatingWebhookConfiguration updates the specified mutating webhook configuration
+func UpdateValidatingWebhookConfiguration(c client.Client, updated *admissionregistrationv1.ValidatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitMedium, func() (bool, error) {
+		existing, err := GetValidatingWebhookConfiguration(c, updated.Name)
+		if err != nil {
+			klog.Errorf("Error getting ValidatingWebhookConfiguration: %v", err)
+			return false, nil
+		}
+		if err := c.Patch(context.TODO(), existing, client.MergeFrom(updated)); err != nil {
+			klog.Errorf("error patching ValidatingWebhookConfiguration object %q: %v, retrying...", updated.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// IsMutatingWebhookConfigurationMatched expects a matching MutatingWebhookConfiguration to be present in the cluster
+func IsMutatingWebhookConfigurationMatched(c client.Client) bool {
+	if err := wait.PollImmediate(RetryShort, WaitLong, func() (bool, error) {
+		existing, err := GetMutatingWebhookConfiguration(c, DefaultMutatingWebhookConfiguration.Name)
+		if err != nil {
+			klog.Errorf("Error getting MutatingWebhookConfiguration: %v", err)
+			return false, nil
+		}
+
+		equal := equality.Semantic.DeepDerivative(DefaultMutatingWebhookConfiguration.Webhooks, existing.Webhooks)
+		if !equal {
+			klog.Infof("MutatingWebhookConfiguration is not yet equal, retrying...")
+		}
+		return equal, nil
+	}); err != nil {
+		klog.Errorf("Error waiting for match with expected MutatingWebhookConfigurationMatched: %v", err)
+		return false
+	}
+	return true
+}
+
+// IsValidatingWebhookConfigurationMatched expects a matching MutatingWebhookConfiguration to be present in the cluster
+func IsValidatingWebhookConfigurationMatched(c client.Client) bool {
```

Comment:
Maybe this should be called `Synced` rather than `Matched`, since we are checking if MAO has synced it to what we expect right?